services/secret-service/server/src/main/java/org/apache/airavata/mft/secret/server/backend/file/FileBasedSecretBackend.java

```diff
@@ -211,4 +213,57 @@ public boolean updateAzureSecret(AzureSecretUpdateRequest request) throws Except
     public boolean deleteAzureSecret(AzureSecretDeleteRequest request) throws Exception {
         throw new UnsupportedOperationException("Operation is not supported in backend");
     }
+
+    @Override
+    public Optional<GCSSecret> getGCSSecret(GCSSecretGetRequest request) throws Exception {
+        JSONParser jsonParser = new JSONParser();
+        InputStream inputStream = FileBasedSecretBackend.class.getClassLoader().getResourceAsStream(secretFile);
+
+        try (InputStreamReader reader = new InputStreamReader(inputStream)) {
+            Object obj = jsonParser.parse(reader);
+            JSONArray resourceList = (JSONArray) obj;
+
+            List<GCSSecret> gcsSecrets = (List<GCSSecret>) resourceList.stream()
+                    .filter(resource -> "GCS".equals(((JSONObject) resource).get("type").toString()))
+                    .map(resource -> {
+                        JSONObject r = (JSONObject) resource;
+                        StringBuilder contentBuilder = new StringBuilder();
+                        BufferedReader br = null;
+                        String jsonContents = "";
+                        try {
+                            br = new BufferedReader(new FileReader(r.get("jsonCredentialsFilePath").toString()));
```

Comment:
We read from the file path only in the SecretService. The agent no longer requires the file to be accessible, we assume the file is available with the SecretService.

Reply:
Because this is the file based secret backend, it should read all credentials from the secrets.json file. You can add the credential json file content with a secret id to secrets json.  Then It will be consistent with other secret implementations

For example,
{
"type": "GCS",
"secret_id": "gcs-sample-secret",
"json": <json_content>
}

Reply:
But with this approach a very large json has to be here as a string. Would that be fine?
 
The problem is GCS requires this entire file and it's contents, unlike other providers that can work with a single key.

Reply:
If the backend is the File Based backend, it should read from the secrets.json. That's the idea. You can format the json to a single line and put it here. If you read the json form an external file path, it breaks the purpose of the file based backend.

Reply:
Okay we will work on it.